### PR TITLE
sublime3: fix hardcoded /bin/bash when executing commands for build systems

### PR DESCRIPTION
**Pull request by mogria**
###### Motivation for this change

See Issue #12011. You can now run Builds from within Sublime Text.

###### Things done

- [x] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] Linux
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/applications/editors/sublime3/default.nix [hunk 1]
```diff
@@ -33,6 +33,21 @@ in let
...
 
+    # make exec.py in Default.sublime-package use "/usr/bin/env bash"
+    # instead of "/bin/bash"
+    patchPhase = ''
+      mkdir Default.sublime-package-fix
+      ( cd Default.sublime-package-fix
+        ${unzip}/bin/unzip ../Packages/Default.sublime-package > /dev/null
+        substituteInPlace "exec.py" --replace \
+          '["/bin/bash"' \
+          '["/usr/bin/env", "bash"'
```
Comment (Mic92) [inline pkgs/applications/editors/sublime3/default.nix]: how about?

```
substituteInPlace "exec.py" --replace \
  '["/bin/bash"' \
  '["${pkgs.bash}/bin/bash"'

Comment (Mic92) [inline pkgs/applications/editors/sublime3/default.nix]: Just noticed that you have already revert this change.

File: pkgs/applications/editors/sublime3/default.nix [hunk 2]
```diff
@@ -33,15 +33,15 @@ in let
...
+    # an LD_PRELOAD instead of "/bin/bash"
     patchPhase = ''
       mkdir Default.sublime-package-fix
       ( cd Default.sublime-package-fix
         ${unzip}/bin/unzip ../Packages/Default.sublime-package > /dev/null
         substituteInPlace "exec.py" --replace \
-          '["/bin/bash"' \
-          '["/usr/bin/env", "bash"'
+          "[\"/bin/bash\"" \
+          "[\"$out/sublime_bash\""
```
Comment (Mic92) [inline pkgs/applications/editors/sublime3/default.nix]: Would not my solution has the same effect?

Comment (mogria) [inline pkgs/applications/editors/sublime3/default.nix]: So is this fine? As you've said, It's already revised in the second commit.

**Comment by mogria** [comment]
Is there any issue with this? Can this be merged?

**Comment by Mic92** [comment]
Thanks!
